blog/2020-09-25-gh-release.md

```diff
@@ -0,0 +1,165 @@
+---
+title: "Designing GitHub workflows for automated releases"
+author: thomas
+author_url: https://github.com/thomaseizinger
+author_image_url: https://avatars1.githubusercontent.com/u/5486389
+tags: [github,automation,gitflow,releases]
+---
+
+Applied in the right circumstances, using GitFlow as your branching model can make life significantly easier.
+The COMIT team decided to adopt GitFlow for [`comit-rs`](https://github.com/comit-network/comit-rs) quite some time ago because it allows development to continue while a release is in progress, hence removing friction from our development flow.
+
+This post gives a quick overview of how we are using GitHub actions to automate most of aspects around drafting releases.
+
+<!--truncate-->
+
+When drafting a release, there is two things we want to have control over:
+
+1. The next version number
+2. The Git revision
+
+Everything else around drafting a release can be implied from these two pieces of information.
+Good automation gives you control over what you care about and hides all the other stuff from you.
+
+The [`RELEASING.md`](https://github.com/comit-network/comit-rs/blob/dev/RELEASING.md#technical-documentation) document in the repository already explains some of the design aspects around the process we implemented but there are a few things that I would like to place emphasis on.
+
+## Using a dedicated bot user
+
+[By default](https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret), GitHub actions injects an access token into your workflows as `secrets.GITHUB_TOKEN`.
+This token is used to authenticate against the GitHub API and shows up as the `github-actions[bot]` user.
+This is quite convenient but also has limitations.
+Rightfully, GitHub is [preventing](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#triggering-new-workflows-using-a-personal-access-token) you from creating recursive workflows.
+For example, if one of your workflows pushes a new branch, a CI workflow usually triggered by the `push` event will not run.
+
+This is one of the reasons why we are using a dedicated [bot user](https://github.com/comit-botty-mc-botface) for our releases.
+In GitFlow, drafting a new release starts by creating a release branch like `release/0.10.0`.
+Our automation does this for us, pushes the branch and opens a PR against `master` (remember that in GitFlow, development happens on a `dev` or `develop` branch).
+If we would be using the standard `GITHUB_TOKEN` for this, our CI workflow would not be triggered for the release branch.
+That is a pity because obviously, we would like to verify that the revision we are about to release actually works and should hence be tested!
```

Comment:
obviously ... actually 

and the next sentence has another actually again.

For my taste bit too much (...)ly

Reply:
Well, that is English grammar :see_no_evil: